libraries/botbuilder-azure/botbuilder/azure/blob_storage.py

```diff
@@ -63,24 +62,31 @@ def __init__(self, settings: BlobStorageSettings):
         return items
 
     async def write(self, changes: Dict[str, object]):
+        if changes is None:
+            raise Exception("Changes are required when writing")
+        if not changes:
+            return
+
         self.client.create_container(self.settings.container_name)
         self.client.set_container_acl(
             self.settings.container_name, public_access=PublicAccess.Container
         )
 
-        for name, item in changes.items():
-            e_tag = (
-                None if not hasattr(item, "e_tag") or item.e_tag == "*" else item.e_tag
-            )
-            if e_tag:
-                item.e_tag = e_tag.replace('"', '\\"')
+        for (name, item) in changes.items():
+            e_tag = item.e_tag if hasattr(item, "e_tag") else item.get("e_tag", None)
+            e_tag = None if e_tag == "*" else e_tag
+            if e_tag == "":
```

Comment:
if not e_tag

Reply:
This was intentional based on [this test](https://github.com/microsoft/botbuilder-dotnet/blob/master/tests/Microsoft.Bot.Builder.Tests/StorageBaseTests.cs#L155). If the e_tag is None, it should act as wildcard. If the e_tag is specified as an empty string, it shouldn't work.